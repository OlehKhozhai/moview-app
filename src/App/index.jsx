import React, { memo } from "react";

import styles from "./styles.module.scss";

const App = () => {
  return <div className={styles.root}>App</div>;
};

export default memo(App);
