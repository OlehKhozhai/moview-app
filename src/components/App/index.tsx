import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import ErrorBoundary from 'components/_common/ErrorBoundary';
import MainRouter from 'routing';
import { store } from 'redux/store';
import styles from './styles.module.scss';

const App: React.FC = () => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <ErrorBoundary>
          <div className={styles.root}>
            <MainRouter />
          </div>
        </ErrorBoundary>
      </BrowserRouter>
    </Provider>
  );
};

export default App;
