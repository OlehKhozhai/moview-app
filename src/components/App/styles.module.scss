.root {
  text-align: center;
  padding: 20px;
}

.title {
  color: white;
}
