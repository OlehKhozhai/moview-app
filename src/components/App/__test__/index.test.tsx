import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';

import App from '../index';

describe('App component', () => {
  test('should render App component by default on home page', () => {
    const component = create(
      <MemoryRouter initialEntries={['/']}>
        <App />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
