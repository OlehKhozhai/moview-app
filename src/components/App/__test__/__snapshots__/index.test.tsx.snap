// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render App component by default on home page 1`] = `
<div
  className="root"
>
  <div
    className="root"
  >
    <div
      className="header"
    >
      <a
        className="root"
        href="/"
        onClick={[Function]}
      >
        <span
          className="bold"
        >
          netflix
        </span>
        roulete
      </a>
      <button
        className="root secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        + add movie
      </button>
    </div>
    <h2
      className="title"
    >
      Find your movie
    </h2>
    <div
      className="root"
    >
      <input
        className="input"
        onChange={[Function]}
        placeholder="What do you want to watch?"
        value=""
      />
      <button
        className="root primary button"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>
  <hr
    className="root"
  />
  <nav
    className="root"
  >
    <ul
      className="root"
    >
      <li
        className="root active"
        onClick={[Function]}
      >
        all
      </li>
      <li
        className="root"
        onClick={[Function]}
      >
        documentary
      </li>
      <li
        className="root"
        onClick={[Function]}
      >
        comedy
      </li>
      <li
        className="root"
        onClick={[Function]}
      >
        horror
      </li>
      <li
        className="root"
        onClick={[Function]}
      >
        crime
      </li>
    </ul>
    <div
      className="root"
    >
      <span
        className="label"
      >
        Sort By
      </span>
      <div
        className="root"
      >
        <span
          className="value"
          onClick={[Function]}
        >
          release date
        </span>
      </div>
    </div>
  </nav>
  <div
    className="root"
  >
    <h2
      className="noMovieFound"
    >
      No movies found
    </h2>
  </div>
  <footer
    className="root"
  >
    <a
      className="root"
      href="/"
      onClick={[Function]}
    >
      <span
        className="bold"
      >
        netflix
      </span>
      roulete
    </a>
  </footer>
</div>
`;
